packages/ui/FloatingLayer/FloatingLayer.js

```diff
@@ -193,27 +197,30 @@ class FloatingLayerBase extends React.Component {
 		const {floatLayerClassName} = this.props;
 		const floatingLayer = this.context.getFloatingLayer();
 
-		if (!this.node && floatingLayer && typeof document !== 'undefined') {
-			invariant(
-				this.context.getFloatingLayer,
-				'FloatingLayer cannot be used outside the subtree of a FloatingLayerDecorator'
-			);
+		if (this.node || !floatingLayer) return;
```

Comment:
don't we need to check for `typeof document !== 'undefined'` for snapshot?

Reply:
It's only necessary in module-scoped code and in the constructor (and componentWillMount). `componentDidMount` and `componentWillReceiveProps` aren't called during pre-render.